File: app/src/main/java/com/wavecat/inline/Command.java
package com.wavecat.inline;

import androidx.annotation.NonNull;

import org.luaj.vm2.LuaValue;

public class Command {

    private final String category;
    private final LuaValue callable;
    private final String description;

    public Command(String category, LuaValue callable, String description) {
        this.category = category;
        this.callable = callable;
        this.description = description;
    }

    @SuppressWarnings("unused")
    public String getCategory() {
        return category;
    }

    public String getDescription() {
        return description;
    }

    public LuaValue getCallable() {
        return callable;
    }

    @NonNull
    @Override
    public String toString() {
        return "Command{" +
                "category='" + category + '\'' +
                ", callable=" + callable +
                ", description='" + description + '\'' +
                '}';
    }
}

File: app/src/main/java/com/wavecat/inline/service/commands/Command.kt
package com.wavecat.inline.service.commands

import org.luaj.vm2.LuaValue

/**
 * Represents a command that can be executed.
 *
 * @property category The category of the command. Can be null if the command doesn't belong to a specific category.
 * @property callable The Lua function that will be executed when the command is invoked.
 * @property description A brief description of what the command does. Can be null if no description is provided.
 */
data class Command(
    val category: String?,
    val callable: LuaValue,
    val description: String?,
)